src/casync-http.c

```diff
@@ -11,67 +12,1037 @@
+                return -EIO;
+        }
+
+        c = curl_easy_setopt(handle, CURLOPT_PROTOCOLS,
+                             arg_protocol == PROTOCOL_FTP ? CURLPROTO_FTP :
+                             arg_protocol == PROTOCOL_SFTP? CURLPROTO_SFTP:
+                             CURLPROTO_HTTP | CURLPROTO_HTTPS);
+        if (c != CURLE_OK) {
+                log_error_curle(c, "Failed to limit CURL protocols to HTTP/HTTPS/FTP/SFTP");
+                return -EIO;
+        }
+
+        if (IN_SET(arg_protocol, PROTOCOL_HTTP, PROTOCOL_HTTPS)) {
+                /* Default since libcurl 7.62.0 */
+                c = curl_easy_setopt(handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_2_0);
+                if (c != CURLE_OK)
+                        log_error_curle(c, "Failed to set CURL HTTP version to HTTP2, ignoring");
+
+                c = curl_easy_setopt(handle, CURLOPT_PIPEWAIT, 1L);
+                if (c != CURLE_OK)
+                        log_error_curle(c, "Failed to set CURL pipewait, ignoring");
+        }
+
+        if (arg_protocol == PROTOCOL_SFTP) {
+                /* activate the ssh agent. For this to work you need
+                   to have ssh-agent running (type set | grep SSH_AGENT to check) */
+                c = curl_easy_setopt(handle, CURLOPT_SSH_AUTH_TYPES, CURLSSH_AUTH_AGENT);
+                if (c != CURLE_OK)
+                        log_error_curle(c, "Failed to turn on CURL SSH agent support, ignoring");
+        }
+
+        if (arg_rate_limit_bps > 0) {
+                c = curl_easy_setopt(handle, CURLOPT_MAX_SEND_SPEED_LARGE, arg_rate_limit_bps);
+                if (c != CURLE_OK) {
+                        log_error_curle(c, "Failed to set CURL send speed limit");
+                        return -EIO;
+                }
+
+                c = curl_easy_setopt(handle, CURLOPT_MAX_RECV_SPEED_LARGE, arg_rate_limit_bps);
+                if (c != CURLE_OK) {
+                        log_error_curle(c, "Failed to set CURL receive speed limit");
+                        return -EIO;
+                }
+        }
+
+        c = curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, write_callback);
+        if (c != CURLE_OK) {
+                log_error_curle(c, "Failed to set CURL callback function");
+                return -EIO;
+        }
+
+        c = curl_easy_setopt(handle, CURLOPT_WRITEDATA, write_data);
+        if (c != CURLE_OK) {
+                log_error_curle(c, "Failed to set CURL callback data");
+                return -EIO;
+        }
+
+        if (private) {
+                c = curl_easy_setopt(handle, CURLOPT_PRIVATE, private);
+                if (c != CURLE_OK) {
+                        log_error_curle(c, "Failed to set CURL private data");
+                        return -EIO;
+                }
+        }
+
+        /* (void) curl_easy_setopt(handle, CURLOPT_SSL_VERIFYPEER, false); */
+
+        /* (void) curl_easy_setopt(handle, CURLOPT_VERBOSE, 1L); */
```

Comment:
Didn't do it, as `-v` is already an option to enable casync verbosity. Maybe `-vv` to enable curl verbose as well?

Since there's already https://github.com/systemd/casync/issues/194 on the same topic, I would batch that altogether. I have some WIP regarding this issue, but I wanted to wait until this PR is completed and merged before finishing it and opening a PR.

Reply:
#217 propagates both options `--log-level` and `-v` from `casync` to `casync-http`. The cURL verbosity is enabled for log-level `notice` and higher (i.e. on `debug`).